cli/bench/bench.go

```diff
@@ -39,39 +49,62 @@ func incrementRequest(err error, results *Results) {
 }
 
 // requestsLoop continuously executes the insert query until the benchmark time runs out.
-func requestsLoop(ctx context.BenchCtx, tarantoolConnection *tarantool.Connection, results *Results, backgroundCtx bctx.Context) {
+func requestsLoop(requestsSequence *RequestsSequence, backgroundCtx bctx.Context) {
 	for {
 		select {
 		case <-backgroundCtx.Done():
 			return
 		default:
-			_, err := tarantoolConnection.Exec(
-				tarantool.Insert(
-					benchSpaceName,
-					[]interface{}{common.RandomString(ctx.KeySize), common.RandomString(ctx.DataSize)}))
-			incrementRequest(err, results)
+			request := requestsSequence.getNext()
+			request.operation(&request)
 		}
 	}
 }
 
-// connectionLoop runs "ctx.SimultaneousRequests" requests execution threads through the same connection.
-func connectionLoop(ctx context.BenchCtx, tarantoolConnection *tarantool.Connection, results *Results, backgroundCtx bctx.Context) {
+// connectionLoop runs "ctx.SimultaneousRequests" requests execution goroutines
+// through the same connection.
+func connectionLoop(
+	ctx *context.BenchCtx,
+	requestsSequence RequestsSequence,
```

Comment:
Why `RequestsSequence` pass by value and not by pointer?

Reply:
RequestsSequence is created along with the connection to reduce the cost of mutexes. If you pass a pointer, then you need to create an array of sequences, which does not make sense